### DISCUSSION TITLE
Extracting all heads from modflow

### DISCUSSION THREAD

**Discussion by zargol7419** [modflowpy/flopy#1579] [Q&A] [answered]
Hi everybody
I need to exctrat all calculated heads from modflow using Flopy. Because I need every head in every cell of the modflow's map for my project. How can I extract all of them? I have used the code ( get-data()) to extract heads but I have received only a portion of required heads in specific locations but I need all of calculated heads in every cell. Is there any code to extract this information by using flopy? Or I should just work with the files which have been created by running modflow model?
Thanks in advance for your answers. They can be really helpful

**Comment by dbrakenhoff** [comment, marked as answer]
Hi @zargol7419, try this?

```python
hobj = flopy.utils.HeadFile("mymodel.hds")
heads = hobj.get_alldata()   # dimensions (ntimes, nlay, nrow, ncol)
```

**Comment by emorway-usgs** [comment]
@zargol7419 This question looks nearly the same as #1567.  In trying to understand what might be different from your other question, make sure you set the output times in Output Control.  If you're expecting output at a specific simulation time and don't see it, it may be because Output Control has not been set up correctly.
